src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.cs

```diff
@@ -43,6 +43,14 @@ internal sealed override bool OnTryRead(ref Utf8JsonReader reader, Type typeToCo
                     ThrowHelper.ThrowJsonException_DeserializeUnableToConvertValue(TypeToConvert);
                 }
 
+                if (state.Current.ParentProperty?.TryPopulate(ref state) == true)
```

Comment:
1. This one attempts to change state.Current.ReturnValue
2. It also handles case where `Get` returns null

Reply:
to clarify, here we check if we should attempt to populate while in the beginning we are already in the middle of populating (since we're not actually gonna use continuation logic from parametrized ctor we're using base class logic)